File: election-frontend/src/interfaces/IPostService.ts
import {IPost} from "@/interfaces/IPost";
import {IPageResponse} from "@/interfaces/IPageResponse";

export interface IPostService {
    getById(id: number): Promise<IPost>;
    getAll(sort: string, page: number, size: number, query?: string): Promise<IPageResponse<IPost>>;
}

File: election-backend/src/main/java/com/example/electionbackend/interfaces/IPostService.java
package com.example.electionbackend.interfaces;

import com.example.electionbackend.dto.PostDTO;
import com.example.electionbackend.dto.PostDetailDTO;
import com.example.electionbackend.model.Post;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Service interface for managing Post entities.
 */
public interface IPostService {
    PostDetailDTO getById(int id);
    Page<PostDetailDTO> getAll(Pageable pageable, String query);
    PostDTO create(PostDTO postDTO);
    Post update(Post postDTO);
    void delete(int id);
}
